src/Umbraco.Web.UI.Client/src/views/propertyeditors/grid/grid.controller.js

```diff
@@ -343,6 +343,37 @@ angular.module("umbraco")
             }
         };
 
+        var shouldApply = function(item, itemType, gridItem) {
+            if (item.applyTo === undefined || item.applyTo === null || item.applyTo === "") {
+                return true;
+            }
+
+            if (typeof (item.applyTo) === "string") {
+                return item.applyTo === itemType;
+            }
+
+            if (itemType === "row") {
+                if (item.applyTo.row === undefined) {
+                    return false;
+                }
+                if (item.applyTo.row === null || item.applyTo.row === "") {
+                    return true;
+                }
```

Comment:
Defining the row (meaning you actually typed row in the config .. whether you specify it as null or an empty string, will mean it should be shown on rows (like the old row string property). Not actually specifying which rows to show but simply putting it in there, would result in it applying to all types of rows like it did with just the row string (return true).